internal/router/router.go

```diff
@@ -23,36 +24,68 @@ import (
 var ErrClusterEmpty = xerrors.Wrap(fmt.Errorf("cluster empty"))
 
 type router struct {
-	config            config.Config
-	balancerConfig    routerconfig.Config
+	driverConfig      config.Config
+	routerConfig      routerconfig.Config
 	pool              *conn.Pool
 	discovery         discovery.Client
 	discoveryRepeater repeater.Repeater
+	localDCDetector   func(ctx context.Context, endpoints []endpoint.Endpoint) (string, error)
 
 	m                sync.RWMutex
 	connectionsState *connectionsState
 }
 
-func (r *router) clusterDiscovery(ctx context.Context) error {
+func (r *router) clusterDiscovery(ctx context.Context) (err error) {
+	var discoveryTrace trace.DriverRouterDiscoveryInfo
+	start := time.Now()
+	defer func() {
+		discoveryTrace.Error = err
+		discoveryTrace.Latency = time.Since(start)
+
+		trace.DriverOnRouterDiscovery(
```

Comment:
```
        var (
           endpoints []endpoint.Endpoint
           start := time.Now()
        }
	defer func() {
               trace.DriverOnRouterDiscovery(
 			r.driverConfig.Trace(),
 			time.Since(start),
 			endpoints,
 			r.routerConfig.DetectlocalDC,
 			discoveryTrace.LocalDC,
 			err,
 		)
	}()
	// main logic
```